openstack/sharedfilesystems/v2/shares/results.go

```diff
@@ -144,3 +144,35 @@ func (r GetExportLocationsResult) Extract() ([]ExportLocation, error) {
 	err := r.ExtractInto(&s)
 	return s.ExportLocations, err
 }
+
+// GrantAccess contains all information associated with an OpenStack share Grant Access
+type GrantAccess struct {
```

Comment:
`AccessRight` works for me as well. I did not consider how nuanced the words in this PR could be, so I apologize about that.

One thing to keep in mind is that Manila has an API call to list access rules, which is a list of all `GrantAccess` calls that have been performed. Upon first glance, the body of these entries are slightly different than what `AccessRight` would be. Of course, it's always possible the docs are wrong.

https://developer.openstack.org/api-ref/shared-file-systems/?expanded=list-access-rules-detail#list-access-rules

If that API call gets implemented and new result structs need created, it might be a good idea to carry the `AccessRight` naming pattern to be consistent. Perhaps `AccessRightEntries` which is made up of individual `AccessRightEntry` (assuming `AccessRightEntry` is different than `AccessRight`).

Just thinking ahead. No action needs made.